store/bsdb/prefix.go

```diff
@@ -108,16 +108,14 @@ func (b *BsDBImpl) filterObjects(nodes []*SlashPrefixTreeNode) ([]*ListObjectsRe
 		}
 	}
 
-	for i := 0; i < ObjectsNumberOfShards; i++ {
-		err = b.db.Table(GetObjectsTableNameByShardNumber(i)).
-			Where("object_id in (?)", objectIDs).
-			Find(&objects).Error
-		//stop after finding one set?
-		if err != nil {
-			return nil, err
-		}
-		totalObjects = append(totalObjects, objects...)
+	err = b.db.Table(GetObjectsTableName(bucketName)).
```

Comment:
As all objects for the same buckets will only be stored into the same single sharded object table, only 1 query against GetObjectsTableName(bucketName) will be enough.